Modifications of .lua file to add numbering and comment callout

SarahMuse opened the pull request:
The changes made to the unilur.lua file are: 

1. I have added a callout for "Comments". It can be used to show how to give points (like for an exam) or for any other comments needed. This is why I gave a general naming. The comments will show together with solutions if they are used. No extra document is created. This was something that I did not like with the assign extension, because for me solutions and comments (or points) belong to the same document.
2. I have changed the numbering of the exercises. The default needed to have a header to have 1.1, 1.2, etc... Otherwise, it was 0.1, 0.2, ... I never liked that behavior, so I took the opportunity to modify that. Thus, I also add the numbering to solutions and comments. I thought that it might be useful, especially for lenghty documents or when students print pdf.

SarahMuse committed:
Modifications of .lua file to add numbering and comment callout

SarahMuse committed:
Update version and contributor

SarahMuse committed:
Update NEWS.md with the last commit information

SarahMuse commented:
Thank you for your clear steps! I have done the first three parts. I will update the `example.qmd` during the week.

What do you mean with "I added a comment for creating a custom comment callout."? I did not see anything anywhere. Just for my understanding, you would like to have the possibility to custom the text and the color of the comment callout?

ginolhac commented:
thanks @SarahMuse!

What I did is a review in your `unilur.lua` code L94-97. I am pasting it for clarity:

>we should create a new type like comment that we define in `unilur.css`. This custom callout will avoid overwriting the warning and we can also change the colour to something we like.
see how I did it for solutions [here](https://github.com/ginolhac/unilur/commit/d5356b16dac1d82622246af9dd6129a1884f55a4)

This is to avoid this situation where the comment callout looks like a warning when close-by.


<img width="760" height="341" alt="image" src="https://github.com/user-attachments/assets/904a6ba8-b0c1-4d10-bb71-93681e74fc82" />

You can then pick up the colour you want, I am not very good at this so it helps if you come up with a suggestion :slightly_smiling_face:

ginolhac committed:
for PDF format use custom tcolorbox for both solution and comment

SarahMuse committed:
Custom tcolorbox to mimic the callouts visual and change the color of comment callout to grey

SarahMuse committed:
Update the callout of comment callout to grey

SarahMuse commented:
Thank you! This looks great! I customized a bit further the tcolorbox in order to look almost identical to the callouts. I added a lot of comments in there too (so it is easier for anyone to customize if needed). 
I also change the color of the comment callout to grey. 
If it is ok for you, I will then move on with the documentation. Unless, you have further suggestions

SarahMuse committed:
Modify the tip callout color to yellow for .tex file

SarahMuse committed:
Add comments in the .css file for the tip callout

SarahMuse committed:
Updated version of README with additional information, comment callout and numbering

SarahMuse commented:
Thx! 
Oh! Now, I get it why the tip callout color was changed in the .css file. I forgot to ask you! I am totally fine with it! And it does make sense to respect the original {unilur}! I have made the changes in the .tex file and just added comment in the .css (so everyone can understand the change of color).

I have revised the README file. I made several changes. Please feel free to accept/decline any of them. Here are the main points:

1. I updated the current release of Quarto. If you are ok, just delete the sentence in bracket
2. I made a more detailed text about what unilur extension is (thinking about people not knowing it). I would suggest to delete the text "The current package does more than the hiding/highlighting of solution code chunks for teaching practicals", as I do not think that it does help to understand what the extension does.
3. I added a bit more details for the installation 
4. For the usage, I tried to simplify but also add information. I kept in mind that I will create a vignette with more information. I deleted the part about `show-solution` as I am not using it at all and I am not sure if it should be in the README, but more in the vignette (if you want to keep it). 

I tried my best to keep your README template and add additional information to help users about the extension. Let me know what you think and what you want to keep or not. I am quite open. And sorry for having made so many changes!

SarahMuse committed:
Small correction to the table Solutions

ginolhac committed:
using github callout and redo screenshot

SarahMuse commented:
Thank you to you! It was a great learning opportunity for me! 
I will then create a new PR for the vignette. I will try my best to have it quite soon.
